plugins/vpc-eni/network/hnswrapper/interface_windows.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package hnswrapper
+
+import (
+	"encoding/json"
+	"net"
+
+	"github.com/Microsoft/hcsshim"
+)
+
+var (
+	// HnsDefaultMinVersion is the minimum version of HNS on Windows instances supported by the plugin.
+	HnsDefaultMinVersion = hcsshim.HNSVersion1803
+)
+
+// WindowsNetworkBuilder provides an interface for building the HNS networks and endpoints.
+type WindowsNetworkBuilder interface {
+	CheckHNSVersion(hnsMinVersion hcsshim.HNSVersion) error
+	FindOrCreateHNSNetwork(nw *HNSNetwork) (bool, error)
+	DeleteHNSNetwork(nw *HNSNetwork) error
+	FindOrCreateHNSEndpoint(ep *HNSEndpoint) (bool, error)
+	DeleteHNSEndpoint(ep *HNSEndpoint) error
+	AttachEndpoint(ep *HNSEndpoint) error
+	CreatePolicy(ep *HNSEndpoint, policy interface{}) error
+}
+
+// HNSNetwork is the configuration of the HNS network.
+type HNSNetwork struct {
```

Comment:
There are too many nested levels of indirection here. I'm lost. Why do we have to define our own HNS structures here? Why can't we use the ones in the official hcsshim package?

Reply:
We can discuss this offline during our meeting tomorrow.